core/modules/filter/filter.pages.inc

```diff
@@ -656,14 +656,16 @@ function _filter_url_has_protocol($url) {
 function _filter_get_file_id($href) {
   $result = 0;
 
-  // Check URI for valid file scheme.
-  if (!file_valid_uri($href)){
+  // Make sure this is a file.
+  global $base_path;
+  $file_path = $base_path . config_get('system.core', 'file_public_path');
```

Comment:
Why are we assuming that the public files scheme is used for files? ...file uploads in CKEditor can be configured to use the private files directory:

![image](https://user-images.githubusercontent.com/2423362/198524992-e6d2eeb0-1544-46e0-bfa4-6a7b45032fbb.png)

Reply:
Oops, you're right. We have to consider that.